apps/condo/domains/billing/schema/SumBillingReceiptsService.js

```diff
@@ -4,72 +4,91 @@
+        },
         {
             access: true,
             type: 'type BillingReceiptsSumOutput { sum: String! }',
         },
     ],
-    
     queries: [
         {
             access: access.canSumBillingReceipts,
-            schema: '_allBillingReceiptsSum (where: BillingReceiptWhereInput!): BillingReceiptsSumOutput',
-            resolver: async (parent, args, context, info, extra = {}) => {
-                const { where } = args
+            schema: '_allBillingReceiptsSum (where: BillingReceiptsSumInput!): BillingReceiptsSumOutput',
+            resolver: async (parent, args) => {
+                const { where: { period, tin, organizationId, importRemoteSystem } } = args
 
-                // We do not want to give user an easy way to query whole database. You must specify period in where query.
-                //
-                // It was done like this, and not as a separate argument for several reasons:
-                // 1. in order to keep validation of period in one place - in GraphQL
-                // 2. in order to keep all<model> styled api arguments somewhat consistent (e.x _allPaymentsSum / allBillingReceipts )
-                //
-                // If you try to add more periods to the where clause, like: where: (period=a1 period=a2) you will get an error
-                // If you try to be hacky and do this: where: (period=a1 period_gt=a1) you are going to have intersection of these clauses which means you will get only period=a1 ones!
-                if (!where.period || typeof where.period !== 'string') {
-                    throw new GQLError(ERRORS.PERIOD_BADLY_SPECIFIED)
+                if (!tin && !organizationId) {
+                    throw new GQLError(ERRORS.TIN_OR_ORGANIZATION_ID_MUST_BE_SPECIFIED)
                 }
 
-                const billingReceiptsLoader = new GqlWithKnexLoadList({
-                    listKey: 'BillingReceipt',
-                    fields: 'id toPay',
-                    where,
-                    sortBy: 'createdAt_DESC',
+                const { keystone } = getSchemaCtx('BillingReceipt')
+                const { knex } = getDatabaseAdapter(keystone)
+
+                const organizationWhere = { deletedAt: null }
+                if (organizationId) {
+                    organizationWhere.id = organizationId
+                }
+                if (tin) {
+                    organizationWhere.tin = tin
+                }
+                if (importRemoteSystem) {
+                    organizationWhere.importId_not = null
+                    organizationWhere.importRemoteSystem = importRemoteSystem
+                }
+
+                const contextIds = []
+                await allItemsQueryByChunks({
+                    schemaName: 'BillingIntegrationOrganizationContext',
+                    where: {
+                        organization: organizationWhere,
+                        deletedAt: null,
+                    },
```

Comment:
Should we take into account the integration status (`InProgress`/`Error`/`Finished`)? Only `Finished`?

Reply:
It does not matter in this case, It just should exist and be not deleted